### ISSUE TITLE
integrate_profile does not take lowest bin into account

### ISSUE THREAD

**Issue by BerendWijers** [adokter/bioRad#543]
Hi all,

@bart1 and I have dug deeper and we believe we've found a bug which (can) cause large numerical differences for VIR calculations.

As a follow-up from #542. 

It turns out that in integrate_profile in version 0.6.1 a delta height `dh` vector is being calculated, instead of using 0.5.2's `index`. With the calculation of `dh` the lowest bin is discarded whereas we expect it to be included, as `alt_min >= 0`. The 0.5.2's `index` does include the lowest bin with equal `alt_min` value. 

The example below shows why it hasn't been picked up in any tests. The `example_vp` has no density information in the lowest bin. Therefore, any changes to the inclusion of the lowest bin would not be reflected in the resulting values. Adding density to the lowest bin _should_ yield differences but does not. 
 
``` r
require(bioRad)
#> Loading required package: bioRad
#> Welcome to bioRad version 0.6.1
#> Docker daemon running, Docker functionality enabled (vol2bird version 0.5.0.9169)
a<-example_vp
a$data$dens[1]<-100
all.equal(
  integrate_profile(example_vp),
  integrate_profile(a)
)
#> [1] TRUE
sessionInfo()
#> R version 4.1.2 (2021-11-01)
#> Platform: x86_64-pc-linux-gnu (64-bit)
#> Running under: Ubuntu 22.04.1 LTS
#> 
#> Matrix products: default
#> BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3
#> LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so
#> 
#> locale:
#>  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
#>  [3] LC_TIME=nl_NL.UTF-8        LC_COLLATE=en_US.UTF-8    
#>  [5] LC_MONETARY=nl_NL.UTF-8    LC_MESSAGES=en_US.UTF-8   
#>  [7] LC_PAPER=nl_NL.UTF-8       LC_NAME=C                 
#>  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
#> [11] LC_MEASUREMENT=nl_NL.UTF-8 LC_IDENTIFICATION=C       
#> 
#> attached base packages:
#> [1] stats     graphics  grDevices utils     datasets  methods   base     
#> 
#> other attached packages:
#> [1] bioRad_0.6.1
#> 
#> loaded via a namespace (and not attached):
#>  [1] Rcpp_1.0.9          lattice_0.20-45     tidyr_1.2.1        
#>  [4] png_0.1-7           assertthat_0.2.1    digest_0.6.30      
#>  [7] utf8_1.2.2          R6_2.5.1            plyr_1.8.7         
#> [10] reprex_2.0.2        evaluate_0.18       spam_2.9-1         
#> [13] httr_1.4.4          ggplot2_3.3.6       highr_0.9          
#> [16] pillar_1.8.1        RgoogleMaps_1.4.5.3 rlang_1.0.6        
#> [19] curl_4.3.3          rstudioapi_0.14     data.table_1.14.4  
#> [22] raster_3.6-3        R.utils_2.12.2      R.oo_1.25.0        
#> [25] rmarkdown_2.18      styler_1.8.1        rgdal_1.5-32       
#> [28] foreign_0.8-82      stringr_1.4.1       munsell_0.5.0      
#> [31] compiler_4.1.2      xfun_0.34           pkgconfig_2.0.3    
#> [34] htmltools_0.5.3     tidyselect_1.2.0    tibble_3.1.8       
#> [37] gridExtra_2.3       codetools_0.2-18    fansi_1.0.3        
#> [40] viridisLite_0.4.1   dplyr_1.0.10        withr_2.5.0        
#> [43] rhdf5filters_1.6.0  bitops_1.0-7        R.methodsS3_1.8.2  
#> [46] grid_4.1.2          gtable_0.3.1        lifecycle_1.0.3    
#> [49] magrittr_2.0.3      scales_1.2.1        cli_3.4.1          
#> [52] stringi_1.7.8       viridis_0.6.2       fs_1.5.2           
#> [55] sp_1.5-0            generics_0.1.3      vctrs_0.5.0        
#> [58] Rhdf5lib_1.16.0     tools_4.1.2         ggmap_3.0.1        
#> [61] R.cache_0.16.0      glue_1.6.2          purrr_0.3.5        
#> [64] maps_3.4.1          fields_14.1         jpeg_0.1-9         
#> [67] fastmap_1.1.0       yaml_2.3.6          terra_1.6-17       
#> [70] colorspace_2.0-3    rhdf5_2.38.1        maptools_1.1-5     
#> [73] dotCall64_1.0-2     knitr_1.41
```

<sup>Created on 2022-12-05 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by adokter** [comment]
Hi @bart1 @BerendWijers yes this bug sneaked in recently, should be resolved in latest dev version.

We've run into a licensing issue with one of the libraries used by vol2bird that enables reading IRIS format, which has been holding up vol2birdR CRAN release, which has been holding up bioRad release. I'll give it a bit more time I hear you a bioRad release fixing this issue is important.

**Comment by adokter** [comment]
already fixed in #533
